apps/src/code-studio/components/progress/StatusProgressDot.jsx

```diff
@@ -62,14 +66,31 @@ export const StatusProgressDot = React.createClass({
       }
     }
 
-    return (
-      <ProgressDot
-        level={level}
-        courseOverviewPage={courseOverviewPage}
-        stageId={stageId}
-        status={status}
-      />
-    );
+    if (experiments.isEnabled('progressBubbles')) {
+      const onCurrent = currentLevelId &&
+        ((level.ids && level.ids.map(id => id.toString()).indexOf(currentLevelId) !== -1) ||
+        level.uid === currentLevelId);
```

Comment:
I don't remember all the details offhand (I agree this is pretty hairy and might be improveable). FWIW, this was copied over from here https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/progress/ProgressDot.jsx#L269